### PR TITLE
Allow for Additional OverRideParams similar to Appcmd's implementation…

### PR DESCRIPTION
**Pull request by michaeljprentice**
…n.  To handle complex strings that came up in testing with a large number of -setParam's, I implemented Encoding and Decoding with Base64.  The Guid and Name was altered to make it easy to test with side by side.

### CODE REVIEW

**Comment by bryanmacfarlane** [comment]
Assigning to Muthu to get some traction

File: Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1 [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
+	[string]$ServerNameIndication,
+	[string]$SslCertThumbPrint,
+	[string]$AppPoolName,
+	[string]$DotNetVersion,
+	[string]$PipeLineMode,
+	[string]$AppPoolIdentity,
+	[string]$AppPoolUsername,
+	[string]$AppPoolPassword,
+	[string]$AppCmdCommands,
+	[string]$SkipAction,
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1]: What is this SkipAction?

File: Tasks/IISWebAppDeployment/DeployIISWebApp.ps1 [hunk 2]
```diff
@@ -36,15 +37,15 @@ $hostName = [string]::Empty
 
 if($protocol -eq "http")
 {
-    $hostName = $hostNameWithHttp
+	$hostName = $hostNameWithHttp
 }
 elseif($serverNameIndication -eq "true")
 {
-    $hostName = $hostNameWithSNI
+	$hostName = $hostNameWithSNI
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/DeployIISWebApp.ps1]: Merge your repo with latest code from vso-agent-tasks

File: Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1 [hunk 3]
```diff
@@ -58,560 +60,622 @@ $MsDeployInstallPathRegKey = "HKLM:\SOFTWARE\Microsoft\IIS Extensions\MSDeploy"
...
+	}
+
+	
+	if(-not (IsInputNullOrEmpty -str $skipAction))
+	{
+		$msDeployCmdArgs = [string]::Format('{0} -skip:{1}', $msDeployCmdArgs, $skipAction)
+	}
+
+	if(-not (IsInputNullOrEmpty -str $overRideParams))
+	{
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1]: Can you help me with your Approach?

File: Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1 [hunk 4]
```diff
@@ -58,560 +60,622 @@ $MsDeployInstallPathRegKey = "HKLM:\SOFTWARE\Microsoft\IIS Extensions\MSDeploy"
...
+		if(-not [string]::IsNullOrEmpty($overRideParam.Trim(' ')))
+		{
+			if($overRideParam -notmatch "Connection String" -and $overRideParam -notmatch "ConnectionString")
+			{
+				[array]$splitOverRide  = $overRideParam.Split('=')
+				$name = $splitOverRide[0].Trim('"').Trim("'").Trim(' ')
+				$value = $splitOverRide[1].Trim('"').Trim("'").Trim(' ')
+				
+				$cleanOverRide = "`"$name`"=`"$value`""
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1]: We are aware of an issue with complex msdeploy params and the issue we are planning to fix in different layer. Currently if that issue is blocking you can use this work around in your repo, but we will fix that in remote deployer layer not in task and then we will remove hack of putting single in place of double quotes.

**Comment by mvvsubbu** [comment]
I am going ahead and closing this PR. Get back to me if you have any issues

**Comment by mvvsubbu** [comment]
Hi Michael,

I am fixing the issue in IIS Task with complex msdeploy arguments. Can you share the exact inputs that were causing problem, so that I can make sure it is fixing your issue as well.

**Pull request closed without merging**